pulp_ostree/tests/functional/api/test_import.py

```diff
@@ -1,210 +1,173 @@
-        artifact = gen_artifact(f"{repo_name1}.tar")
-
-        # 7. commit the tarball to a Pulp repository
-        repo = self.repositories_api.create(OstreeOstreeRepository(**gen_repo()))
-        self.addCleanup(self.repositories_api.delete, repo.pulp_href)
-        commit_data = OstreeImportAll(artifact["pulp_href"], repo_name1)
-        response = self.repositories_api.import_all(repo.pulp_href, commit_data)
-        repo_version = monitor_task(response.task).created_resources[0]
-
-        # 8. check the number of created commits, branches (refs), and objects
-        repository_version = self.versions_api.read(repo_version)
-        added_content = repository_version.content_summary.added
-        self.assertEqual(added_content["ostree.config"]["count"], 1)
-        self.assertEqual(added_content["ostree.refs"]["count"], 2)
-        self.assertEqual(added_content["ostree.commit"]["count"], 3)
-        self.assertEqual(added_content["ostree.object"]["count"], 4)
-
-        # 9. publish the parsed tarball
-        distribution_data = OstreeOstreeDistribution(**gen_distribution(repository=repo.pulp_href))
-        response = self.distributions_api.create(distribution_data)
-        distribution = monitor_task(response.task).created_resources[0]
-        self.addCleanup(self.distributions_api.delete, distribution)
-
-        ostree_repo_path = self.distributions_api.read(distribution).base_url
-
-        # 10. initialize a second local OSTree repository and pull the content from Pulp
-        remote_name = init_local_repo_with_remote(repo_name2, ostree_repo_path)
-        self.addCleanup(shutil.rmtree, Path(repo_name2))
-        validate_repo_integrity(repo_name2, f"{remote_name}:foo")
-
-    def test_single_import_one_commit(self):
-        """Append a new child commit to an existing repository and pull it from Pulp."""
-        self.single_ref_import(1)
-
-    def test_single_import_more_commits(self):
-        """Append a set of commits to an existing repository and pull them from Pulp."""
-        self.single_ref_import(5)
-
-    def single_ref_import(self, number_of_commits):
-        """Import child commits, publish them, and pull them from Pulp."""
-        self.repo_name1 = utils.uuid4()
-        self.repo_name2 = utils.uuid4()
-        sample_dir1 = Path(utils.uuid4())
-        sample_dir2 = Path(utils.uuid4())
-        sample_file1 = sample_dir1 / Path(utils.uuid4())
+def test_simple_tarball_import(
+    artifacts_api_client,
+    gen_object_with_cleanup,
+    monitor_task,
+    ostree_distributions_api_client,
+    ostree_distribution_factory,
+    ostree_repository_factory,
+    ostree_repositories_api_client,
+    ostree_repositories_versions_api_client,
+    tmp_path,
+):
+    """Import a repository consisting of three commit, publish it, and pull it from Pulp."""
+    os.chdir(tmp_path)
+    repo_name1 = str(uuid.uuid4())
+    repo_name2 = str(uuid.uuid4())
+    sample_dir = Path(str(uuid.uuid4()))
```

Comment:
```suggestion
    repo_name1 = str(uuid.uuid4())
    repo_name2 = str(uuid.uuid4())
    sample_dir = tmp_path / str(uuid.uuid4())
```

Reply:
hum... if we remove the _os.chdir_ we will also need to modify the lines that depends on the dir path, right?
For example:
```
subprocess.run(["ostree", f"--repo={repo_name1}", "init", "--mode=archive"])
```
or
```
subprocess.run(["tar", "-cvf", f"{repo_name1}.tar", f"{repo_name1}/"])
```

Reply:
really, only the last change is relevant:
```
sample_dir = tmp_path / str(uuid.uuid4())
```

Reply:
Yeah,you maybe right. (You could probably also add a pwd to the subprocess call.)